flask_dance/consumer/oauth2.py

```diff
@@ -152,6 +161,8 @@ def __init__(
         self.token_url_params = token_url_params or {}
         self.redirect_url = redirect_url
         self.redirect_to = redirect_to
+        self.code_challenge_method = code_challenge_method
+        self._use_pkce = is_valid_code_challenge_method(self.code_challenge_method)
```

Comment:
I don't think we need this. I'd say there are two good ways to structure this API:

1. This `__init__` method accepts both a `use_pkce` boolean (default to False) _and_ a `code_challenge_method` string (default to "S256"). If `use_pkce` is True, then we do PKCE. Otherwise, we don't.
2. This `__init__` method accepts a `code_challenge_method` string (default to None). If it is set to a truthy string, then we do PKCE with the challenge method provided. If it is set to a falsy value, including the default None, then we don't do PKCE at all.

Personally, I think that option 1 is the better one. The spec says that "plain" should be the default, but it also says that devices that _can_ do sha256 hashing MUST do so, so "S256" seems like a reasonable default for the method in this case. But if you prefer option 2, let's discuss it!

Either way, I don't think we need to worry about validating the value of `code_challenge_method`. The spec says that its optional, and [the oauthlib implementation](https://github.com/oauthlib/oauthlib/blob/0adbbe10ed8ef822d1c780987fffc56670ce3f9f/oauthlib/oauth2/rfc6749/clients/base.py#L504) falls back on "plain" if the `code_challenge_method` argument is anything other than "S256", so I don't think we need to raise an exception if it's something unexpected.

Reply:
Here I was not sure what would be the preferred interface for the method. But option number 1 looks good to me, I will update it this way.